src/Integration.Vsix/Analysis/AnalysisConfigMonitor.cs

```diff
@@ -46,20 +48,30 @@ internal sealed class AnalysisConfigMonitor : IAnalysisConfigMonitor, IDisposabl
         [ImportingConstructor]
         public AnalysisConfigMonitor(IAnalysisRequester analysisRequester,
             IUserSettingsProvider userSettingsProvider, // reports changes to user settings.json
```

Comment:
This existing class monitors changes that can impact the analysis results.
The ESLintBridge analyzer classes listen to the `ConfigChanged` event and flag that the ESLintBridge needs to be re-initialised i.e. the rules config re-calculated.

The solution binding changing (e.g. from unbound to bound) does affect the analysis, so I've changed the monitor to listen for this event too.

There is a performance impact to this change, as it will also trigger the re-analysis of all open documents. However, that's probably the behaviour we want.